### PR TITLE
Clarify 'or' requirement in message for subject age + DOB

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
We discovered this during an alpha test and it caused some confusion

DANDI only requires one or the other; theoretically possible to have both, but the message implies it's required to have both, which is not true

### CODE REVIEW

**Comment by garrettmflynn** [comment]
While a better direction for users of the GUIDE, it's technically incorrect—as the metadata _is_ missing age AND date of birth.

Would we be able to note `Subject requires an age or date of birth.` instead? 

Also, the reason it's confusing on the GUIDE is that only one of the fields (`age`) is checked AND that field provides information about both values. Would it be out of the question to expose the two checks separately—possibly combining them for an aggregate check that still behaves like this one?

**Comment by CodyCBakerPhD** [comment]
> While a better direction for users of the GUIDE, it's technically incorrect—as the metadata is missing age AND date of birth.

I've modified the logic to reflect the cases

> Would we be able to note Subject requires an age or date of birth. instead?

That is what is says now, right?

> Would it be out of the question to expose the two checks separately

The check applies to the Subject object entirely, and via PyNWB the form of `date_of_birth` is already checked for format so no need for an extra check function

**Comment by garrettmflynn** [comment]
> That is what is says now, right?

I guess when I read "Subject is missing age or date_of_birth.", I think _one is provided_ but the other is not. But this PR was raised because "Subject is missing age and date_of_birth." implies that both _are / could be_ required—though it is correctly describing the Subject metadata.

What both of these messages fail to accomplish is a clear indication of the user's course of action based on the requirements. So I was thinking a message that directly states "Subject requires an age or date of birth." is a better way to describe _what the user should do_ (provide one or the other) rather than merely describe _the status of the fields being evaluated_ by the Inspector.

File: src/nwbinspector/checks/nwbfile_metadata.py [hunk 1]
```diff
@@ -132,11 +132,10 @@ def check_doi_publications(nwbfile: NWBFile):
 @register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=Subject)
 def check_subject_age(subject: Subject):
     """Check if the Subject age is in ISO 8601 or our extension of it for ranges."""
-    if subject.age is None:
-        if subject.date_of_birth is None:
-            return InspectorMessage(message="Subject is missing age and date_of_birth.")
-        else:
-            return
+    if subject.age is None and subject.date_of_birth is None:
+        return InspectorMessage(message="Subject is missing age or date_of_birth.")
```
Comment (CodyCBakerPhD) [inline src/nwbinspector/checks/nwbfile_metadata.py]:
```suggestion
        return InspectorMessage(message="Subject is missing age or date_of_birth. Please specify at least one.")
```

File: tests/unit_tests/test_nwbfile_metadata.py [hunk 2]
```diff
@@ -336,7 +336,7 @@ def test_pass_check_subject_age_with_dob():
 def test_check_subject_age_missing():
     subject = Subject(subject_id="001")
     assert check_subject_age(subject) == InspectorMessage(
-        message="Subject is missing age and date_of_birth.",
+        message="Subject is missing age or date_of_birth.",
```
Comment (CodyCBakerPhD) [inline tests/unit_tests/test_nwbfile_metadata.py]:
```suggestion
        message="Subject is missing age or date_of_birth. Please specify at least one.",
```

**Comment by CodyCBakerPhD** [comment]
Except the word 'requirement' in the message is only true with the DANDI config, not in general (anyone can make a Subject object without anything more than an ID) - and making config-dependent error messages would be more complicated than simply correcting the wordage here

> a better way to describe what the user should do (provide one or the other) rather than merely describe the status of the fields being evaluated by the Inspector.

The phrasing across the Inspector is consistently the latter; it only reports what is missing, or what structures are written incorrectly; occasionally it will mention how to fix it, but for existence checks I'd it is pretty self-evident that if the Inspector says something is missing, the way to get it to dismiss the message is to make it not missing

In this special case there are two things that are each missing in order to trigger the message, and the incorrect message is to say that both are missing and therefore both need to be specified to make the message go away

Anyway, I added an extra clarification to the message without making it seem like a hard requirement

**Comment by garrettmflynn** [comment]
Ah thank you for the clarification about the difference between _when_ these requirements hold.

With your clarification, though, I think it's clearer to keep the `and` over the `or` since that's the actual structure of the data when it comes in. But the extra sentence helps!

File: src/nwbinspector/checks/nwbfile_metadata.py [hunk 3]
```diff
@@ -132,11 +132,10 @@ def check_doi_publications(nwbfile: NWBFile):
 @register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=Subject)
 def check_subject_age(subject: Subject):
     """Check if the Subject age is in ISO 8601 or our extension of it for ranges."""
-    if subject.age is None:
-        if subject.date_of_birth is None:
-            return InspectorMessage(message="Subject is missing age and date_of_birth.")
-        else:
-            return
+    if subject.age is None and subject.date_of_birth is None:
+        return InspectorMessage(message="Subject is missing age or date_of_birth. Please specify at least one.")
```
Comment (CodyCBakerPhD) [inline src/nwbinspector/checks/nwbfile_metadata.py]:
```suggestion
        return InspectorMessage(message="Subject is missing age and date_of_birth. Please specify at least one of these fields.")
```

File: tests/unit_tests/test_nwbfile_metadata.py [hunk 4]
```diff
@@ -336,7 +336,7 @@ def test_pass_check_subject_age_with_dob():
 def test_check_subject_age_missing():
     subject = Subject(subject_id="001")
     assert check_subject_age(subject) == InspectorMessage(
-        message="Subject is missing age and date_of_birth.",
+        message="Subject is missing age or date_of_birth. Please specify at least one.",
```
Comment (CodyCBakerPhD) [inline tests/unit_tests/test_nwbfile_metadata.py]:
```suggestion
        message="Subject is missing age and date_of_birth. Please specify at least one of these fields.",
```

**Comment by CodyCBakerPhD** [comment]
@garrettmflynn OK, back to the old message plus an extra response on how to resolve

**Review by garrettmflynn — APPROVED** [review]
Looks good to me!

**Pull request merged**